internal/pkg/service/common/distribution/assigner.go

```diff
@@ -39,6 +39,13 @@ func (a *Assigner) Nodes() []string {
 	return out
 }
 
+// NodesCount method returns count of known nodes.
+func (a *Assigner) NodesCount() int {
+	a.mutex.RLock()
+	defer a.mutex.RUnlock()
+	return len(a.nodes.Hosts())
+}
+
```

Comment:
For tests.

Reply:
BTW: the `distribution` package has good documentation and is important, did you @jachym-tousek-keboola  had time to study it?

Reply:
Note: `distribution.GroupNode` has a listener, to watch distribution changes, but here it is not needed.

If a source node fails, less data will be written in some slices than in others, but that is not a problem. The load balancer will divide the work between remaining source nodes.

A listener/logic is necessary if it is necessary to take over work after a failed node, but here the k8s load balancer solves it.